testing/xfpga/test_enum_c.cpp

```diff
@@ -198,14 +377,29 @@ TEST_P(enum_c_p, function) {
   EXPECT_EQ(num_matches_, 0);
 }
 
+/**
+ * @test       socket_id
+ *
+ * @brief      When the filter socket_id is set and it is valid,
+ *             the function returns the number of resources that
+ *             match that socket_id.
+ */
 TEST_P(enum_c_p, socket_id) {
   auto device = platform_.devices[0];
   ASSERT_EQ(fpgaPropertiesSetSocketID(filter_, device.socket_id), FPGA_OK);
   EXPECT_EQ(
       xfpga_fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(), &num_matches_),
       FPGA_OK);
-  EXPECT_EQ(num_matches_, 2);
+  EXPECT_EQ(num_matches_, GetNumFpgaTypes() * 2);
```

Comment:
I don't follow what this test is doing. socket_id is 0 on pac systems, and can be 0, 1, ... on integrated systems.

Reply:
Right, and as the test infrastructure is currently implemented, the test will run one test for each platform detected on the system. The current platform socket will always be 0, MCP or DCP, and we enumerate for that. So we expect that fpgaEnumerate() will return 2 matches for MCP and, if detected, 2 matches for DCP.